### PR TITLE
LPD-28414 Index Asset Vocabulary + Categories ERC

### PR DESCRIPTION
**Pull request by joshuacords**
**Story:** [LPD-28414](https://liferay.atlassian.net/browse/LPD-28414)
**Subtasks:**
- [LPD-31073](https://liferay.atlassian.net/browse/LPD-31073)
- [LPD-33425](https://liferay.atlassian.net/browse/LPD-33425) Tests

**Author:** @joshuacords

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
ci:report:59214597

File: modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/search/index/contributor/test/AssetCategoryDocumentContributorTest.java [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+			_assetVocabularyLocalService.addVocabulary(
+				TestPropsValues.getUserId(), _group.getGroupId(),
+				RandomTestUtil.randomString(), _serviceContext);
+
+		assetVocabulary.setVisibilityType(visibilityType);
+
+		assetVocabulary = _assetVocabularyLocalService.updateAssetVocabulary(
+			assetVocabulary);
+
+		_assetVocabularyList.add(assetVocabulary);
```
[the remark this answers is not in the document]
Comment (joshuacords) [reply modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/search/index/contributor/test/AssetCategoryDocumentContributorTest.java:229]: The assetVocabularies are stored here that way the test will delete them during cleanup.

File: modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/search/index/contributor/test/AssetCategoryDocumentContributorTest.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+			booleanQuery.addMustQueryClauses(
+				_queries.term(fieldName, externalReferenceCode));
+		}
+
+		countSearchRequest.setQuery(booleanQuery);
+
+		CountSearchResponse countSearchResponse = _searchEngineAdapter.execute(
+			countSearchRequest);
+
+		Assert.assertTrue(countSearchResponse.getCount() == 1);
```
[the remark this answers is not in the document]
Comment (joshuacords) [reply modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/search/index/contributor/test/AssetCategoryDocumentContributorTest.java:305]: All the other ones that B Chan reviewed were assertTrue, so I think let's just let it be.

File: modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/search/index/contributor/test/AssetCategoryDocumentContributorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuacords) [reply modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/search/index/contributor/test/AssetCategoryDocumentContributorTest.java:229]: I checked it now, I think it was the case we had to do this years ago but now we don't.

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
Made almost all of those changes @gustavolimav, take another look and let me know.

Thank you!

**Review by gustavolimav — APPROVED** [review]
Now LGTM! 🚀

**Comment by gustavolimav** [comment]
ci:forward:force

**Pull request closed without merging**